script/check_icon_use.sh

```diff
@@ -0,0 +1,498 @@
+    "rotorua_lakes_council_nz.py"
+    "roundlookup_uk.py"
+    "roundrocktexas_gov.py"
+    "runnymede_gov_uk.py"
+    "rushcliffe_gov_uk.py"
+    "rushmoor_gov_uk.py"
+    "rv_de.py"
+    "ryde_nsw_gov_au.py"
+    "salford_gov_uk.py"
+    "samiljo_se.py"
+    "sammelkalender_ch.py"
+    "sandnes_no.py"
+    "sbazv_de.py"
+    "scambs_gov_uk.py"
+    "scheibbs_umweltverbaende_at.py"
+    "seattle_gov.py"
+    "sepan_remondis_pl.py"
+    "sheffield_gov_uk.py"
+    "shellharbourwaste_com_au.py"
+    "sholland_gov_uk.py"
+    "shropshire_gov_uk.py"
+    "sica_lu.py"
+    "sivom_rivedroite_fr.py"
+    "snaga_mb_si.py"
+    "solihull_gov_uk.py"
+    "south_norfolk_and_broadland_gov_uk.py"
+    "southampton_gov_uk.py"
+    "southderbyshire_gov_uk.py"
+    "southglos_gov_uk.py"
+    "southkesteven_gov_uk.py"
+    "southtyneside_gov_uk.py"
+    "stadt_bamberg_de.py"
+    "staedteservice_de.py"
+    "staffordbc_gov_uk.py"
+    "stalbans_gov_uk.py"
+    "stavanger_no.py"
+    "stevenage_gov_uk.py"
+    "stirling_uk.py"
+    "stirling_wa_gov_au.py"
+    "stockport_gov_uk.py"
+    "stockton_gov_uk.py"
+    "stoke_gov_uk.py"
+    "stonnington_vic_gov_au.py"
+    "stratford_gov_uk.py"
+    "stroud_gov_uk.py"
+    "sunderland_gov_uk.py"
+    "sutton_gov_uk.py"
+    "swansea_gov_uk.py"
+    "swindon_gov_uk.py"
+    "sysav_se.py"
+    "tameside_gov_uk.py"
+    "tauranga_govt_nz.py"
+    "tbv_velbert_de.py"
+    "tekniskaverken_se.py"
+    "telford_gov_uk.py"
+    "tewkesbury_gov_uk.py"
+    "tkeliai_lt.py"
+    "tmbc_gov_uk.py"
+    "tonnenleerung_de.py"
+    "toogoodtowaste_co_nz.py"
+    "toronto_ca.py"
+    "townsville_qld_gov_au.py"
+    "tunbridgewells_gov_uk.py"
+    "umweltverbaende_at.py"
+    "unley_sa_gov_au.py"
+    "uttlesford_gov_uk.py"
+    "valeofglamorgan_gov_uk.py"
+    "vasyd_se.py"
+    "vestfor_dk.py"
+    "victoriapark_wa_gov_au.py"
+    "vivab_se.py"
+    "vmeab_se.py"
+    "walsall_gov_uk.py"
+    "wanneroo_wa_gov_au.py"
+    "warrington_gov_uk.py"
+    "warwickdc_gov_uk.py"
+    "was_wolfsburg_de.py"
+    "wastecollection_mt.py"
+    "wastenet_org_nz.py"
+    "waverley_gov_uk.py"
+    "wealden_gov_uk.py"
+    "welhat_gov_uk.py"
+    "wellington_govt_nz.py"
+    "wermelskirchen_de.py"
+    "west_dunbartonshire_gov_uk.py"
+    "west_norfolk_gov_uk.py"
+    "westberks_gov_uk.py"
+    "westlothian_gov_uk.py"
+    "westnorthants_gov_uk.py"
+    "westoxon_gov_uk.py"
+    "westsuffolk_gov_uk.py"
+    "whitehorse_vic_gov_au.py"
+    "whittlesea_vic_gov_au.py"
+    "wigan_gov_uk.py"
+    "wiltshire_gov_uk.py"
+    "winterthur_ch.py"
+    "wirral_gov_uk.py"
+    "wokingham_gov_uk.py"
+    "wollondilly_nsw_gov_au.py"
+    "wollongongwaste_com_au.py"
+    "wychavon_gov_uk.py"
+    "wyndham_vic_gov_au.py"
+    "wyre_gov_uk.py"
+    "wyreforestdc_gov_uk.py"
+    "yarra_ranges_vic_gov_au.py"
+    "york_gov_uk.py"
+)
+
+# Build a git-exclude and git-include strings
+grep_exclude=""
+grep_include=""
+for excluded in "${whitelist[@]}"
```

Comment:
This loop (and above list of excluded files) could be replaced by adding a specific string or in each source file that shall be excluded by check and then replace the loop with the following:

```sh
for excluded in $(git grep --name "CUSTOM_ICON_WHITELIST = True" -- $checkpath); do
    base=$(basename $excluded)
    grep_exclude="${grep_exclude} :(exclude)*${base}"
    grep_include="${grep_include} ${checkpath}${base}"
done
```